### PR TITLE
feat: flush all tables on shutdown

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

   - call shutdown on ctrl-c signal.
   - invoke flush_tables in datanode instance's shutdown

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/datanode/src/instance.rs [hunk 1]
```diff
@@ -233,13 +235,51 @@ impl Instance {
...
+                .into_iter()
+                .map(|request| self.sql_handler.execute(request, QueryContext::arc())),
+        )
+        .await
+        .map_err(BoxedError::new)
+        .context(ShutdownInstanceSnafu);
+        info!("flush success: {}", flush_result.is_ok());
+        flush_result?;
+
+        Ok(())
```
Comment (evenyag) [inline src/datanode/src/instance.rs:280]: Could we combine these two lines?
```suggestion

        flush_result
```

Comment (waynexia) [reply src/datanode/src/instance.rs:280]: cannot do that, types doesn't match (`()` vs `Vec<_>`)

**Review by evenyag — APPROVED** [review]

File: src/datanode/src/instance.rs [hunk 2]
```diff
@@ -233,13 +235,51 @@ impl Instance {
...
+            .collect::<Vec<_>>();
+        let flush_result = futures::future::try_join_all(
+            flush_requests
+                .into_iter()
+                .map(|request| self.sql_handler.execute(request, QueryContext::arc())),
+        )
+        .await
+        .map_err(BoxedError::new)
+        .context(ShutdownInstanceSnafu);
+        info!("flush success: {}", flush_result.is_ok());
```
Comment (killme2008) [inline src/datanode/src/instance.rs:277]:
```suggestion
        info!("Flushed all tables result: {}", flush_result.is_ok());
```

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**